pkg/pmem-device-manager/pmd-ndctl.go

```diff
@@ -54,19 +49,23 @@ func NewPmemDeviceManagerNdctl() (PmemDeviceManager, error) {
 		}
 	}
 
-	return &pmemNdctl{
-		ctx: ctx,
-	}, nil
+	return &pmemNdctl{}, nil
 }
 
 func (pmem *pmemNdctl) GetCapacity() (map[string]uint64, error) {
 	ndctlMutex.Lock()
 	defer ndctlMutex.Unlock()
 
+	ndctx, err := ndctl.NewContext()
```

Comment:
In all remaining cases (if we agree we can delete use of mutex in NewPmemDeviceManagerNdctl()),
there is both "get lock" and "create context" as pair of operations, so can we perhaps combine those 2 in same function, and use defer in other functions to call another common release-delete function ?
Not a big deal though, its quite small and clear already now.

Reply:
I was debating that with myself and decided to go with separate code for mutex and context because it is clearer and more flexible.